cfme/utils/appliance/__init__.py

```diff
@@ -1964,6 +1964,21 @@ def update_server_roles(self, changed_roles):
         self.server_roles = server_roles
         return server_roles == self.server_roles
 
+    def wait_for_server_roles(self, server_roles, delay=10, timeout=300):
+        """Waits for the server roles to be set
+
+         Args:
+            server_roles: list of server roles to be checked
+            delay: Delay to pass to wait_for (default ```10```)
+            timeout: Number of seconds to wait until timeout (default ``300``)
+         """
+        result = True
+        for role in server_roles:
+            if not wait_for(lambda: self.server_roles[role] is True, delay=delay, timeout=timeout):
```

Comment:
This behavior seems a bit odd, wait_for is going to raise an exception if it times out, not return False.

So the method will either raise an exception if one of the roles times out, or return True.